internal/app/machined/pkg/runtime/v1alpha1/platform/metal/url/variable.go

```diff
@@ -28,31 +28,29 @@ type Variable struct {
 	r     *regexp.Regexp
 }
 
-// AllVariables is a list of all supported variables.
-func AllVariables() []*Variable {
-	return []*Variable{
-		{
-			Key:        constants.UUIDKey,
-			MatchOnArg: true,
-			Value:      UUIDValue(),
-		},
-		{
-			Key:   constants.SerialNumberKey,
-			Value: SerialNumberValue(),
-		},
-		{
-			Key:   constants.MacKey,
-			Value: MACValue(),
-		},
-		{
-			Key:   constants.HostnameKey,
-			Value: HostnameValue(),
-		},
-		{
-			Key:   constants.CodeKey,
-			Value: CodeValue(),
-		},
-	}
+// AllVariables is a map of all supported variables.
+var AllVariables = map[string]*Variable{
```

Comment:
as `Variable` embeds `Value` which contains matched value, we should always generate a fresh (empty) list here, so this should be still a `func()`

Reply:
got it, done.